File: 7_multiple_inheritance.cpp
#include<bits/stdc++.h>
using namespace std;
//parent class
class Vehical
{
    public:
        string brand;

        Vehical(string brand)
        {
            this->brand = brand;
        }

        void VehicalInfo()
        {
            cout<<"Vehical Brand :- "<<brand<<endl;
        }
};
// parent class 
class Engine
{
    public:
        int horsepower;

        Engine(int horsepower)
        {
            this->horsepower = horsepower;
        }

        void EngineInfo()
        {
            cout<<"Engine Horsepower :- "<<horsepower<<endl;
        }
};
//child class , vehical ebong Engine er thake inherit kortaca 
// multiple inheritance ekti class ke ekadhik parent class thake attributes ebong method inherite korta dei
class Car : public Vehical, public Engine
{
    public:
        Car(string brand, int horsepower) : Vehical(brand), Engine(horsepower)
        {

        }
};
int main()
{
    Car car("BMW", 300);
    car.VehicalInfo();
    car.EngineInfo();
    return 0;
}

File: 7_multiple_inheritance.py

class Vehical:
    def __init__(self, brand):
        self.brand = brand

    def vehicalInfo(self):
        print("vehical brand :- ",self.brand)

class Engine:
    def __init__(self, horsepower):
        self.horsepower = horsepower

    def engineInfo(self):
        print("Engine Horsepower :- ", self.horsepower, "HP")


class Car(Vehical, Engine): # multiple inheritance: ek ta class jodi 2 ba tar ceye besi class er property inherite kore tokon take multiple inheritance bole. ei khan e Car (Vehical r Engine) er thake property inherit korca
    def __init__(self, brand, hoursepower):
        Vehical.__init__(self, brand)
        Engine.__init__(self, hoursepower)

car = Car("BMW", 300)
car.vehicalInfo()
car.engineInfo()

print(Car.mro()) # priotity bujar built in method holo : mro() - method resolution order



